api/v3/Case/Getdetails.php

```diff
@@ -52,10 +58,21 @@ function civicrm_api3_case_getdetails($params) {
     if (!is_array($params['case_manager'])) {
       $params['case_manager'] = array('=' => $params['case_manager']);
     }
-    \Civi\CCase\Utils::joinOnManager($sql);
+    \Civi\CCase\Utils::joinOnRelationship($sql, 'manager');
     $sql->where(CRM_Core_DAO::createSQLFilter('manager.id', $params['case_manager']));
   }
 
+  // Add clause to search by non manager role and non client
+  if (!empty($params['contact_involved'])) {
```

Comment:
so what happens if i pass both `case_manager` and `contact_involved` parameters?

Reply:
I have fixed the API so that it gives correct result for
```cv api Case.getcaselist contact_involved=123 case_manager=123``` 
but
not for
```cv api Case.getcaselist contact_involved=123 case_manager={"!=":123} `````
So I dig deep and tried the same combination with case client param instead of contact involved to see the results.
So I tried
```cv api Case.getcaselist contact_id=123 case_manager=123```
and
```cv api Case.getcaselist contact_id=123 case_manager={"!=":123}``` 
and it returned the same pattern of results. On digging deep I found
```cv api Case.getcaselist contact_involved=123 case_manager={"!=":123}``` 
is missing cases where the contact is not the first client. Once I change the contact as first client to the case then it improves ( fix) the response. So I think there is a problem with case_manager param (SQL query) which needs time to be fixed
So, `contact_involved` and `case_manager` works perfect independently, but doesn’t work well with each other (only when case_manager is negated)
Now, as per the comment of Jamie on C51-318
https://compucorp.atlassian.net/browse/C51-318?focusedCommentId=77539&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-77539
> Cases I’m involved with means any case - including those where I am the case manager.

It seems that we don’t need a combination of these two for now and keep them as it.
I have already spent much time on debugging this and can spend more, but I have other tickets to finish and then I have to hop on on another project from tomorrow on. So, I think we can keep the functionality as it is for now and fix the issue later.
What are your thoughts? Also, please let me know if you need a call on it.